### PR TITLE
Feat/new notifcation table design

### PR DESCRIPTION
**Pull request by Jonopono123**
#### Work done
- New design of notification table component
- Added ability to filter notifications based on read status and created date
- Added pagination

#### Tests
Does this PR include tests for the changes introduced?
- [x] Yes
- [ ] No, I found it difficult to test
- [ ] No, they are not required for this change

### CODE REVIEW

File: server/data/notifications.js [hunk 1]
```diff
@@ -114,15 +114,19 @@ const updateNotificationQuery = `
   WHERE "notificationUid" = :nuid
 `;
 
-exports.selectNotificationByEstablishment = async (establishmentUid, limit, offset) =>
-  db.query(selectEstablishmentNotifications, {
+//TODO: This query is failing somewhere
```
Comment (RobertM905) [inline server/data/notifications.js:117]: Is this still failing?

File: src/app/features/notifications/notification-list/notification-list.component.html [hunk 2]
```diff
@@ -1,33 +1,110 @@
...
+              <input
+                class="govuk-checkboxes__input"
+                id="checkbox-{{ i }}"
+                name="checkbox-{{ i }}"
+                type="checkbox"
+                [checked]="allBoxesChecked"
+                data-testid="checkbox-{{ i }}"
+                (click)="pushNotificationToDeleteArray(notification)"
+              />
+              <label class="govuk-label govuk-checkboxes__label govuk-!-padding-0" for="checkbox-{{ i }}"></label>
```
Comment (RobertM905) [inline src/app/features/notifications/notification-list/notification-list.component.html:78]: Is there a label name for each checkbox?

File: src/app/features/notifications/notification-list/notification-list.component.html [hunk 3]
```diff
@@ -1,33 +1,110 @@
...
+            <input
+              class="govuk-checkboxes__input"
+              id="selectAllCheckBoxes"
+              name="selectAllCheckBoxes"
+              type="checkbox"
+              [(ngModel)]="allBoxesChecked"
+              data-testid="selectAllCheckBoxes"
+              (click)="pushAllNotificationsToDeleteArray()"
+            />
+            <label class="govuk-label govuk-checkboxes__label govuk-!-padding-0" for="selectAllCheckBoxes"></label>
```
Comment (RobertM905) [inline src/app/features/notifications/notification-list/notification-list.component.html:50]: Is there a label name for each checkbox?

File: server/data/notifications.js (hunk 1 above)
Comment (Jonopono123) [reply server/data/notifications.js:117]: Nope, it's all good now - I need to remove that comment

File: src/app/features/notifications/notification-list/notification-list.component.html (hunk 2 above)
Comment (RobertM905) [reply src/app/features/notifications/notification-list/notification-list.component.html:78]: As per our discussion. 
The checkbox needs the label to display even without label text.
I am happy with this for now as it's a wider quirk and doesn't need to be addressed here.

**Review by RobertM905 — APPROVED** [review]
Looks good. 
Once you  update that comment on the selectNotificationByEstablishment query.

**Pull request merged**